MFVNeutralino/python/Vertexer_cfi.py

```diff
@@ -46,23 +50,35 @@
+                             resolve_split_vertices_tight = cms.bool(True), # new merging routine 
+                             investigate_merged_vertices = cms.bool(True), # investigate quality cuts on new merged vertices from tight merging 
+                             resolve_shared_jets = cms.bool(True),       # shared-jet mitigation
+                             resolve_shared_jets_src = cms.InputTag('selectedPatJets'), 
                              beamspot_src = cms.InputTag('offlineBeamSpot'),
                              seed_tracks_src = cms.InputTag('mfvVertexTracks', 'seed'),
                              n_tracks_per_seed_vertex = cms.int32(2),
                              max_seed_vertex_chi2 = cms.double(5),
                              use_2d_vertex_dist = cms.bool(False),
                              use_2d_track_dist = cms.bool(False),
                              merge_anyway_dist = cms.double(-1),
-                             merge_anyway_sig = cms.double(3),
+                             merge_anyway_sig = cms.double(4), # merging criteria for loose merging  
                              merge_shared_dist = cms.double(-1),
-                             merge_shared_sig = cms.double(4),
+                             merge_shared_sig = cms.double(4), # default merging shared-track vertices  
                              max_track_vertex_dist = cms.double(-1),
-                             max_track_vertex_sig = cms.double(5),
-                             min_track_vertex_sig_to_remove = cms.double(1.5),
+                             max_track_vertex_sig = cms.double(5), # default track arbitration
+                             min_track_vertex_sig_to_remove = cms.double(1.5), # default track arbitration
                              remove_one_track_at_a_time = cms.bool(True),
                              max_nm1_refit_dist3 = cms.double(-1),
                              max_nm1_refit_distz = cms.double(0.005),
                              max_nm1_refit_count = cms.int32(-1),
+                             trackrefine_sigmacut = cms.double(5), # track refinement criteria
+                             trackrefine_trimmax = cms.double(5), # track refinement criteria 
                              histos = cms.untracked.bool(True),
+                             histos_noshare = cms.untracked.bool(True),   # make plots of no shared-track vertices 
+                             histos_output0 = cms.untracked.bool(True),   # make plots of output vertices after the default vertexing 
+                             histos_output1 = cms.untracked.bool(True),   # make plots of output vertices after the default vertexing  + new merging routine turned on
+                             histos_output2 = cms.untracked.bool(True),   # make plots of output vertices after the default vertexing  + new merging routine turned on  + shared-jet mitigation turned on 
```

Comment:
Do we want all of these, always, or only when doing studies to optimize the merging and shared jet mitigation? Each extra hist that we need to save at this level will add to our disk space usage, so I wonder if we want to turn these off by default?

Also: in Vertexer.cc, it looks like the instantiation of all of the histograms are all controlled by the histos bool. But we only need to create the hists when the relevant bit is on (since even an empty histogram takes up the same amount of space as one that is filled!).

Reply:
I see. You make a good point that empty histograms even if all of these are False can take up some spaces. I think what I can do is to control the creation/define/fill of histograms by the histos bool. 

We don't need all of the histo bools. They are there for future revisits/studies. However, I think we may want to keep histos_output2 on because it has the information of final vertex outputs.

Reply:
Okay! It would be great to only turn on those that we need for general usage (including the adjustment to Vertexer.cc to only instantiate those that are needed for a given configuration).

And it would also be useful to give these names that are more meaningful. `output0` vs `output1` vs `output2` isn't going to be something we remember in a year from now, and the use of words like `default` or `new` in the comment may also cause confusion (since the "new" merging will be our new default!).

Reply:
Okay, I will add the controller of each section by bool and rename the histos bools.